nomenclature/region_mapping_models.py

```diff
@@ -241,29 +248,38 @@ def apply(self, df: IamDataFrame) -> IamDataFrame:
 
                 # Aggregate
                 if self.mappings[model].common_regions is not None:
-                    for cr in self.mappings[model].common_regions:
-                        PYAM_AGG_KWARGS = {
-                            "components",
-                            "method",
-                            "weight",
-                            "drop_negative_weights",
+                    vars = {
+                        var: {
+                            key: value
+                            for key, value in kwargs.items()
+                            if key in PYAM_AGG_KWARGS
                         }
-                        vars = {
-                            var: {
-                                key: value
-                                for key, value in kwargs.items()
-                                if key in PYAM_AGG_KWARGS
-                            }
-                            for var, kwargs in self.definition.variable.items()
-                            if var in model_df.variable
-                            and not kwargs.get("skip-region-aggregation", False)
-                        }
-                        for var, kwargs in vars.items():
-                            agg_df = model_df.aggregate_region(
-                                var,
-                                cr.name,
-                                cr.constituent_regions,
-                                **kwargs,
+                        for var, kwargs in self.definition.variable.items()
+                        if var in model_df.variable
+                        and not kwargs.get("skip-region-aggregation", False)
+                    }
+                    simple_vars = [var for var, kwargs in vars.items() if not kwargs]
```

Comment:
I'd go for a more descriptive name, like `vars_default_args`, and `vars_kwargs`.

Reply:
Done